system-addon/content-src/asrouter/asrouter-content.jsx

```diff
@@ -53,6 +55,45 @@ function shouldSendImpressionOnUpdate(nextProps, prevProps) {
   return (nextProps.message.id && (!prevProps.message || prevProps.message.id !== nextProps.message.id));
 }
 
+function* generateMessages(content) {
+  const cx = new MessageContext("en-US");
+  cx.addMessages(`RichTextSnippet = ${content}`);
+  yield cx;
+}
+
+// Elements allowed in snippet content
+const ALLOWED_TAGS = {
+  b: <b />,
+  i: <i />,
+  u: <u />,
+  strong: <strong />,
+  em: <em />,
+  br: <br />
+};
+
+/**
+ * Transform an object (tag name: {url}) into (tag name: anchor) where the url
+ * is used as href, in order to render links inside a Fluent.Localized component.
+ */
+function convertLinks(links) {
+  if (links) {
+    return Object.keys(links).reduce((acc, linkTag) => {
+      acc[linkTag] = <a href={links[linkTag].url} />;
```

Comment:
Can you whitelist allowed protocols here, so at least `javascript:` isn't allowed?

Reply:
actually you can use `safeURI` here for that